### PR TITLE
Bluetooth: Mesh: optimize performance for lpn node

### PR DESCRIPTION
**Pull request by LingaoM**
Fixed overflow risk when `poll_timeout` is 1s,
although it is not recommended to use it like this,
when it is used like this, `POLL_TIMEOUT_MAX`
will overflow.

When the poll timeout set like above, the default `6`
req_attemps for first pull request lost,
may cause this procedure bigger than poll_timeout,

When lpn node terminate with friend node,
currently will be scan some seconds to receive
`friend clear confirm`, but this will be consume
a lots of energy, so just disable receiver.

Signed-off-by: Lingao Meng <mengabc1086@gmail.com>

### CODE REVIEW

File: subsys/bluetooth/mesh/lpn.c [hunk 1]
```diff
@@ -150,11 +153,6 @@ static void friend_clear_sent(int err, void *user_data)
 {
 	struct bt_mesh_lpn *lpn = &bt_mesh.lpn;
 
-	/* We're switching away from Low Power behavior, so permanently
-	 * enable scanning.
-	 */
-	bt_mesh_scan_enable();
-
```
Comment (jhedberg) [inline subsys/bluetooth/mesh/lpn.c]: Could you move this to a separate commit, please? Also, clarify where the scanning does get enabled if it doesn't happen here.

**Review by jhedberg — CHANGES_REQUESTED** [review]

**Review by jhedberg — CHANGES_REQUESTED** [review]
@LingaoMeng did you forget to add the second commit where you moved the chunk that's now removed from this PR?

**Comment by LingaoMeng** [comment]
> @LingaoMeng did you forget to add the second commit where you moved the chunk that's now removed from this PR?



> @LingaoMeng did you forget to add the second commit where you moved the chunk that's now removed from this PR?

**Comment by LingaoMeng** [comment]
> @LingaoMeng did you forget to add the second commit where you moved the chunk that's now removed from this PR?

when lpn mode enable, the scanning will be start after `FRIEND_REQ_RETRY_TIMEOUT`

File: subsys/bluetooth/mesh/lpn.c (hunk 1 above)
Comment (LingaoMeng) [reply subsys/bluetooth/mesh/lpn.c]: scanning will be enable after `FRIEND_REQ_RETRY_TIMEOUT` seconds.

**Comment by carlescufi** [comment]
@trond-snekvik could you please review?

**Review by trond-snekvik — APPROVED** [review]
LGTM, although I agree with @jhedberg's suggestion of doing the `bt_mesh_scan_enable()` in a separate commit (no need for a separate PR though). This way, you can create more descriptive titles for the two commits than "general performance improvements".

**Review by jhedberg — APPROVED** [review]
